flask/app/analyses.py

```diff
@@ -410,6 +429,11 @@ def update_analysis(id: int):
         "notes",
     ]
 
+    if request.json.get("priority") and request.json.get("priority") != "None":
```

Comment:
Unless there's a reason I might be missing, priority can be added to `editable_columns` by default since all other editable fields are contained in the list regardless of whether there is intent to edit?

Reply:
My thinking here was that "None" isn't a value in the enum, so an additional check is needed so that "None" doesn't get passed to the `mixin` function, which will throw a validation error. If the user wanted to change, say, priority="Clinical" to priority=`null`, either by passing in "None" or `null` as the value, the request would fail without some kind of additional check, or at least it appears that way from my testing. One solution may be to adjust the `mixin` function to accept null as a valid value for nullable enum fields? That way we wouldn't have to pass back 'None' which I agree is not great. But maybe I'm missing a step?

Reply:
Took a stab at this in [e4b9593](https://github.com/ccmbioinfo/stager/pull/522/commits/e4b9593025382ef80a15513aeece84099c059f1e#). I got around the enum check in `mixin` for nullable values by adding a check to see if the column is nullable [here](https://github.com/ccmbioinfo/stager/pull/522/commits/e4b9593025382ef80a15513aeece84099c059f1e#diff-31ba4ef1c8b9ff11c45c7f693530599eeb901ba5bd496a2ace1f5698752d2312R34). It also seems that `mixin` is able to determine column type from the instance only if the field already has a value for that column in the db. I wasn't able to make it work using what I know of introspection functionality, so I just separated the [validation logic](https://github.com/ccmbioinfo/stager/pull/522/commits/e4b9593025382ef80a15513aeece84099c059f1e#diff-dfde89d18ccd5b96d88d26c96d8893c69183bcb28d98a401425b603b891d7701R443) from the [update logic](https://github.com/ccmbioinfo/stager/pull/522/commits/e4b9593025382ef80a15513aeece84099c059f1e#diff-dfde89d18ccd5b96d88d26c96d8893c69183bcb28d98a401425b603b891d7701R448).